odbc/src/api.rs

```diff
@@ -7,11 +11,49 @@ use odbc_sys::{
 
 #[no_mangle]
 pub extern "C" fn SQLAllocHandle(
-    _handle_type: HandleType,
-    _input_handle: Handle,
-    _output_handle: *mut Handle,
-) -> SqlReturn {
-    unimplemented!()
+    handle_type: HandleType,
+    input_handle: Handle,
+    output_handle: *mut Handle,
+) -> SqlReturn {
+    match handle_type {
+        HandleType::Env => {
+            let env = Box::new(EnvHandle::new());
+            env.env.write().unwrap().state = EnvState::Allocated;
```

Comment:
what is the scenario in which we can actually get a handle in the `Unallocated` state? In the Rust code you get this when you create one with `EnvHandle::new()`, but I'm wondering whether it's possible to get to this state via the ODBC C API

Reply:
After `SQLFreeHandle`, but as mentioned, it is an address violation  for anything to actually read that state after `SQLFreeHandle` returns!